rfcs/web_extensions.md

```diff
@@ -0,0 +1,199 @@
+Meaning, a race condition could occur that would cause the `browser.test` assertion results to not
+be reported back to the test harness, since the `browser.test.onMessage` listener wasn't registered.
+To resolve this, user agents would need to queue the results until a listener is registered, and then
+report them.
+
+Last, Since the `Classic` and `BiDi` implementations support loading an extension using a `path` or
+`archivePath` to the extension's resources, the full path to the extension's resources is formulated
+by using the `test.path` before the protocol loads the extension.
+
+A proposed patch is available at https://github.com/web-platform-tests/wpt/pull/50648.
+
+##  `browser.test` API
+
+The `browser.test` API provides utilities for writing and running tests for WebExtensions.
+It includes assertion functions and a test harness to facilitate structured and reliable testing.
+
+
+### Methods
+
+**`browser.test.assertTrue(condition, message)`**\
+Asserts that a given condition is `true`. If the condition is `false`, the test fails.
+
+**Parameters**
+
+* `condition` (boolean) – The condition to assert. If `false`, the test fails.
+* `message` (string, optional) – A message describing the assertion.
+
+\
+**`browser.test.assertFalse(condition, message)`**\
+Asserts that a given condition is `false`. If the condition is `true`, the test fails.
+
+**Parameters**
+
+* `condition` (boolean) – The condition to assert. If `true`, the test fails.
+* `message` (string, optional) – A message describing the assertion.
+
+\
+**`browser.test.assertEq(actual, expected, message)`**\
+Throws an assertion error and fails the test if the actual value is not _logically_ equal to the
+expected value, by these rules:
+
+1) _Plain arrays_ (`Array.isArray()` returns true) are compared for _deep_ equality.
+    1.1) Their `length` and all items are recursively compared by these same rules.
+2) _Plain objects_ (with prototype `Object.prototype` or `null`) are also compared recursively.
+    2.1) Only enumerable _own_ properties are considered, unordered.
+3) Everything else is compared using `Object.is()` [same value algorithm](https://tc39.es/ecma262/#sec-samevalue).
+
+We're starting with support for the minimal set of compound types, and plan to expand custom rules
+to include `ArrayBuffer`s, `Map`s, `Set`s, etc. in congruence with Node's [assert.deepStrictEqual()](https://nodejs.org/api/assert.html#comparison-details_1).
+
+**Parameters**
+
+* `actual` (any) – The actual value.
+* `expected` (any) – The expected value.
+* `message` (string, optional) – A message describing the assertion.
+
+\
+**`browser.test.assertThrows(fn, expectedError, message)`**
+Asserts that a given function throws an error when executed. If the function does not throw, the test fails.
+The test will also fail if `expectedError` is defined and the function does not throw an error with a message matching `expectedError`.
```

Comment:
I think for web-platform-tests `expectedError` must not be used unless the error messages are more standardized than normal JS/DOM exceptions, since otherwise it's embedding browser-specific expectations into the test.

Reply:
Thanks, yeah we don't expect to use this often, or only with a targeted regex for specific keywords.